src/card_operations/card_fetch_challenge.c

```diff
@@ -0,0 +1,182 @@
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "card_fetch_challenge.h"
+
+#include "card_internal.h"
+#include "card_utils.h"
+#include "constant_texts.h"
+#include "flash_api.h"
+#include "nfc.h"
+#include "pow_utilities.h"
+#include "ui_screens.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+extern Wallet_shamir_data wallet_shamir_data;
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+card_error_type_e card_fetch_challenge(const uint8_t *wallet_name) {
+  // X-Coordinate cannot be more than the number of shares
+  if (NULL == wallet_name) {
+    return CARD_OPERATION_DEFAULT_INVALID;
+  }
+
+  card_error_type_e result = CARD_OPERATION_DEFAULT_INVALID;
+  uint8_t target[SHA256_SIZE] = {0};
+  uint8_t random_number[POW_RAND_NUMBER_SIZE] = {0};
+
+  card_operation_data_t card_data = {0};
+  card_data.nfc_data.retries = 5;
+  card_data.nfc_data.init_session_keys = true;
+
+  char heading[50] = "";
+
+  uint8_t wallet_index;
+  get_index_by_name((const char *)wallet_name, &wallet_index);
```

Comment:
return not checked. please add comment accordingly

Reply:
will add assert

Reply:
added